test/functional/output_kafka.go

```diff
@@ -0,0 +1,153 @@
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/ViaQ/logerr/log"
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/test/helpers/kafka"
+	"github.com/openshift/cluster-logging-operator/test/runtime"
+)
+
+const (
+	// Kafka deployment definitions
+	ImageRemoteKafka            = kafka.KafkaImage
+	ImageRemoteKafkaInit        = kafka.KafkaInitUtilsImage
+	kafkaBrokerContainerName    = "broker"
+	kafkaZookeeperContainerName = "zookeeper"
+
+	kafkaInsidePort         = int32(9093)
+	kafkaOutsidePort        = int32(9094)
+	kafkaJMXPort            = int32(5555)
+	zookeeperDeploymentName = "zookeeper"
+
+	zookeeperClientPort         = int32(2181)
+	zookeeperPeerPort           = int32(2888)
+	zookeeperLeaderElectionPort = int32(3888)
+)
+
+func (f *FluentdFunctionalFramework) addKafkaOutput(b *runtime.PodBuilder, output logging.OutputSpec) error {
+	log.V(2).Info("Adding kafka output", "name", output.Name)
+	name := strings.ToLower(output.Name)
+
+	log.V(2).Info("Standing up Kafka instance", "name", name)
+	//steps to deploy kafka single node cluster a. Deploy Zookeeper b. Deploy Broker c. Deploy kafka Consumer
+
+	//step a
+	// Deploy Zookeeper steps : create configmap, create container, create zookeeper service
+
+	//zookeeperm - configmap with zookeeperDeploymentName being "zookeeper", in b.Pod.Namespace
+	zookeepercm := kafka.NewZookeeperConfigMapFunctionalTestPod(b.Pod.Namespace)
+	log.V(2).Info("Creating zookeeper configmap", "namespace", zookeepercm.Namespace, "name", zookeepercm.Name)
+	if err := f.Test.Client.Create(zookeepercm); err != nil {
+		return err
+	}
+
+	//init container for zookeep
+	b.AddInitContainer("init-config0", ImageRemoteKafkaInit).
+		AddVolumeMountToInitContainer("configmapkafka", "/etc/kafka-configmap", "", false, 0).
+		AddVolumeMountToInitContainer("configkafka", "/etc/kafka", "", false, 0).
+		AddVolumeMountToInitContainer("datazookeeper", "/var/lib/zookeeper", "", false, 0).
+		WithCmdArgsToInitContainer([]string{"./bin/bash", "/etc/kafka-configmap/init.sh"}, 0)
+
+	//standup container running zookeeper
+	log.V(2).Info("Adding container", "name", name)
+	b.AddContainer(kafkaZookeeperContainerName, ImageRemoteKafka).
+		AddEnvVar("KAFKA_LOG4J_OPTS", "-Dlog4j.configuration=file:/etc/kafka/log4j.properties").
+		AddContainerPort("client", zookeeperClientPort).
+		AddContainerPort("peer", zookeeperPeerPort).
+		AddContainerPort("leader-election", zookeeperLeaderElectionPort).
+		AddVolumeMount("configmapkafka", "/etc/kafka-configmap", "", false).
+		AddVolumeMount("configkafka", "/etc/kafka", "", false).
+		AddVolumeMount("datazookeeper", "/var/lib/zookeeper", "", false).
+		WithCmd("./bin/zookeeper-server-start.sh /etc/kafka/zookeeper.properties").
+		End().
+		AddConfigMapVolume("configmapkafka", zookeeperDeploymentName).
+		AddEmptyDirVolume("configkafka").
+		AddEmptyDirVolume("zookeeperlogs").
+		AddEmptyDirVolume("datazookeeper")
```

Comment:
I suggest s/"zookeeperlogs"/"zookeeper-logs" and s/"datazookeeper/"zookeeper-data" etc ... so the volume name is prefixed with app ... another option is to drop the "zookeeper" prefixed from the names altogether. As it is now it is confusing.